assets/js/modules/analytics/common/timzezone-select.js

```diff
@@ -0,0 +1,142 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	Select,
+	Option,
+} from '../../../material-components';
+import { countries } from './countries';
+import classnames from 'classnames';
+
+const { default: { country: allCountries } } = countries;
+
+const TimezoneSelect = ( { timezone, setTimezone, hasError } ) => {
+	const [ selectedCountry, setSelectedCountry ] = useState( timezone );
+	const [ selectedTimezoneID, setSelectedTimezoneID ] = useState( '' );
+	let multiTimezone,
+		selectedTimezoneDisplay = selectedTimezoneID,
+		foundTimezone = false;
+	const getTimezoneSelector = () => {
+		const response = (
+			<div >
+				<span>
+					<Select
+						className={ classnames(
+							'googlesitekit-analytics__select-timezone first',
+							{ 'mdc-text-field--error': hasError }
+						) }
+						name="country"
+						enhanced
+						value={ selectedCountry }
+						onEnhancedChange={ ( i, item ) => {
+							setTimezone( item.dataset.value );
+							setSelectedCountry( item.dataset.value );
+							setSelectedTimezoneID( item.dataset.value );
+						} }
+						label={ __( 'Country', 'google-site-kit' ) }
+						outlined
+					>
+						{
+							allCountries && allCountries
+								.map( ( aCountry ) => {
+									// If the selected timezone is in this country, the country should be selected.
+									let value = aCountry.defaultTimeZoneId;
+									const timezoneMatch = aCountry.timeZone.find( ( tz ) => tz.timeZoneId === timezone );
+									if ( timezoneMatch ) {
+										foundTimezone = true;
+										value = timezoneMatch.timeZoneId;
+										if ( aCountry.timeZone.length > 1 ) {
+											multiTimezone = aCountry.timeZone;
+										} else {
+											setSelectedTimezoneID( aCountry.timeZone[ 0 ].displayName );
+											selectedTimezoneDisplay = aCountry.timeZone[ 0 ].displayName;
+											multiTimezone = false;
+										}
+										setSelectedCountry( timezoneMatch.timeZoneId );
+									}
+
+									return (
+										<Option
+											key={ aCountry.displayName }
+											value={ value }
+										>
+											{ aCountry.displayName }
+										</Option>
+									);
+								} ) }
+					</Select>
+				</span>
+				<span>
+					{ multiTimezone
+						? <Select
+							className="googlesitekit-analytics__select-timezone"
+							name="timezone2"
+							enhanced
+							value={ timezone }
+							onEnhancedChange={ ( i, item ) => {
+								setTimezone( item.dataset.value );
+							} }
+							label={ __( 'Timezone', 'google-site-kit' ) }
+							outlined
+						>
+							{
+								multiTimezone
+									.map( ( aTimezone ) => {
+										if ( aTimezone.timeZoneId === timezone ) {
+											foundTimezone = true;
+										}
```

Comment:
Same here - albeit primitive, there shouldn't be any non-mapping logic in here.

Also, the same list of `<Options>` should be reusable here rather than generating it again.

Reply:
Right, I'll remove this part as well. We should only need to regenerate if the user chooses another country.

Reply:
removed this